src/CalculateTipBot/calculateTipBot.ts

```diff
@@ -0,0 +1,38 @@
+import { Bot } from "../shared/types";
+import {
+  ERROR_MESSAGE,
+  FIXED_DECIMAL_AMOUNT,
+  RATES,
+} from "../shared/constants";
+import { calculateTipAmount } from "../shared/utils";
+import { CALCULATE_TIP_START_MESSAGE } from "./constants";
+
+export const addCalculateTipBotCommands = (bot: Bot) => {
+  bot.start((ctx) => ctx.reply(CALCULATE_TIP_START_MESSAGE));
+
+  bot.on("text", (ctx) => {
+    const totalCost = Number(ctx.update.message.text);
+
+    if (!totalCost) {
+      return bot.telegram.sendMessage(ctx.chat.id, ERROR_MESSAGE, {});
+    }
+
+    let ratesMessage = "";
+
+    RATES.forEach((rate) => {
```

Comment:
I could destructure hear but not sure if it would be as readable. I feel like I'd want to rename the properties if I destructured so just going to NOT (for now at least)